src/Estimators/include/BipedalLocomotion/FloatingBaseEstimators/BaseEstimatorFromFootIMU.h

```diff
@@ -0,0 +1,183 @@
+/*
+ * @file BaseEstimatorFromFootIMU.h
+ * @authors Guglielmo Cervettini
+ * @copyright 2023 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the BSD-3-Clause license.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_MATH_BASE_ESTIMATION_FROM_FOOT_IMU_H
+#define BIPEDAL_LOCOMOTION_MATH_BASE_ESTIMATION_FROM_FOOT_IMU_H
+
+#include <memory>
+#include <manif/SE3.h>
+#include <manif/SO3.h>
+#include <iDynTree/KinDynComputations.h>
+#include <iDynTree/Model/Model.h>
+
+#include <BipedalLocomotion/Math/Constants.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/Advanceable.h>
+
+namespace BipedalLocomotion
+{
+    namespace Math
+    {
+
+        /*
+         * BaseEstimatorFromFootIMUState contains the internal state of the estimator,
+         */
+        struct BaseEstimatorFromFootIMUState
+        {
+            manif::SE3d basePose;                //final output of the estimator - pose of the robot root link
+        };
+
+        /*
+         * BaseEstimatorFromFootIMUState contains the input of the BaseEstimatorFromFootIMU class.
+         */
+        struct BaseEstimatorFromFootIMUInput
+        {
+            Eigen::VectorXd jointPositions;      // vector of the robot joint positions
+            Eigen::VectorXd jointVelocities;     // vector of the robot joint velocities
+            manif::SE3d desiredFootPose;         // desired orientation and position of the foot as per footstep planner output
+            manif::SO3d measuredRotation;        // acutal orientation of the foot measured by on-board IMU
+        };
+
+        /*
+         * BaseEstimatorFromFootIMU implements the propagation of the foot pose to the base through the kinematic chain
+         * See [here](https://github.com/ami-iit/element_floatingbase-exteroceptive-estimation/issues/2) for further details.
+         */
+        class BaseEstimatorFromFootIMU
+            : public BipedalLocomotion::System::Advanceable<BaseEstimatorFromFootIMUInput, BaseEstimatorFromFootIMUState>
+        {
+        public:
+            // stance foot reference frame
+            enum class StanceFootRF
+            {
+                l_sole,
+                r_sole
+            };
+
+            /*
+             * Struct holding parameters for the BaseEstimatorFromFootIMU
+             */
+            struct Params
+            {
+                //nothing here at the moment
+            };
+
+            /*
+             * Set the iDynTree::Model of the robot to be used by the estimator.
+             * @return true in case of success, false otherwise.
+             */
+            bool setModel(iDynTree::Model& model);
```

Comment:
The documentation should follow doxygen style. Moreover the model should be passed as a const ref
```suggestion
            /*
             * Set the iDynTree::Model of the robot to be used by the estimator.
             * @param model the model considered in the estimator
             * @return true in case of success, false otherwise.
             */
            bool setModel(const iDynTree::Model& model);
```